pycqed/analysis_v2/coherence_analysis.py

```diff
@@ -907,150 +1197,404 @@ def prepare_plots(self):
+    """
+    # create a parameter set for the initial guess
     p = lmfit.Parameters()
     p.add('slope_ramsey', value=100.0, vary=True)
     p.add('slope_echo', value=100.0, vary=True)
     p.add('intercept', value=100.0, vary=True)
 
-    wrap_residual = lambda p: residual_Gamma(p,
-                                             sensitivity=sensitivity,
-                                             Gamma_phi_ramsey=Gamma_phi_ramsey,
-                                             Gamma_phi_echo=Gamma_phi_echo)
+    # mi = lmfit.minimize(super_residual, p)
+    def wrap_residual(p): return residual_Gamma(
+        p,
+        sensitivity=sensitivity,
+        Gamma_phi_ramsey=Gamma_phi_ramsey,
+        Gamma_phi_echo=Gamma_phi_echo)
     fit_result_gammas = lmfit.minimize(wrap_residual, p)
-    if verbose:
+    if verbose > 0:
         lmfit.printfuncs.report_fit(fit_result_gammas.params)
     return fit_result_gammas
 
 
-def format_value_string(par_name: str, lmfit_par, end_char=''):
+def calc_dephasing_rate(T1, T2):
+    """Calculate pure dephasing rate based on T1 and T2."""
+    gamma = 1/T2 - 1/(2*T1)
+    return gamma
+
+
+def PSD_Analysis(table, freq_resonator=None, Qc=None, chi_shift=None,
+                 path=None):
     """
-    Formats an lmfit par to a  string of value with uncertainty.
+    Power spectral density analysis of transmon coherence.
+
+    Args:
+        table : table containing the data, see below for specification.
+        freq_resonator: readout resonator frequency (in Hz)
+        Qc:             coupling Q of the readout resonator
+        chi_shift       in Hz
+        path:           filepath, if provided is used for saving the plots
+
+
+
+    Input table specification:
+           Row  | Content
+        --------+--------
+            1   | dac
+            2   | frequency
+            3   | T1
+            4   | T2 star
+            5   | T2 echo
+            6   | Exclusion mask (True where data is to be excluded)
+
+    Generates 8 plots:
+        > T1, T2, Echo vs flux
+        > T1, T2, Echo vs frequency
+        > T1, T2, Echo vs flux sensitivity
+        > ratio Ramsey/Echo vs flux
+        > ratio Ramsey/Echo vs frequency
+        > ratio Ramsey/Echo vs flux sensitivity
+        > Dephasing rates Ramsey and Echo vs flux sensitivity
+        > Dac arc fit, use to assess if sensitivity calculation is correct.
+
+    If properties of resonator are provided (freq_resonator, Qc, chi_shift),
+    it also calculates the number of noise photons.
     """
-    val_string = par_name
-    val_string += ': {:.3g}'.format(lmfit_par.value)
-    if lmfit_par.stderr is not None:
-        val_string += r'$\pm$' + '{:.3g}'.format(lmfit_par.stderr)
+    dac, freq, T1, Tramsey, Techo, exclusion_mask = table
+    exclusion_mask = np.array(exclusion_mask, dtype=bool)
+
+    # Extract the dac arcs required for getting the sensitivities
+    fit_result_arch = fit_frequencies(dac, freq, dac0_guess=.1)
```

Comment:
Fails to fit dac arc when I'm using dac that is already in flux quantum units. Needs better dac0_guess. Perhaps based on parabola fit?